### PR TITLE
Implemented openqa test for gd

### PR DESCRIPTION
**Pull request by grisu48**
Implemented openQA test for the [GNU Graphics Library gd](https://libgd.github.io/) in mau-extratests

- Related ticket: https://progress.opensuse.org/issues/49994
- Needles: None
- Verification run: [SLE 15-SP1 [x86_64]](https://openqa.suse.de/t4034774) | | [SLE 15 [x86_64]](https://openqa.suse.de/t4034775) | [SLE 12-SP5 [x86_64]](https://openqa.suse.de/t4034780) |  [SLE 12-SP4 [x86_64]](https://openqa.suse.de/t4034779) | [SLE 12-SP3 [x86_64]](https://openqa.suse.de/t4034778) | [SLE 12-SP2 [x86_64]](https://openqa.suse.de/t4034777) | [SLE 12-SP1 [x86_64]](https://openqa.suse.de/t4034776) | [Tumbleweed](http://phoenix-openqa.qam.suse.de/tests/197) | [Leap](http://phoenix-openqa.qam.suse.de/tests/198)

s390x workers are down ATM

### CODE REVIEW

File: tests/console/gd.pm [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+
+
+use base 'consoletest';
+use strict;
+use warnings;
+use testapi;
+use utils 'zypper_call';
+
+sub run {
+    select_console 'root-console';
```
Comment (foursixnine) [inline tests/console/gd.pm]: I'd use serial console as it would be faster, but it's up to you :)

Comment (grisu48) [reply tests/console/gd.pm]: Good idea! I'll change it to serial console

Comment (grisu48) [reply tests/console/gd.pm]: I think I leave it for now but keep it in mind for the future. Thanks for letting me know

**Comment by okurz** [comment]
Hi grisu48, welcome to this project as first-time contributor :fireworks: :champagne:

Please look into existing commits for the common format of commit messages. The commit message does have some details but could benefit from line wrapping :) Also, please phrase the git commit message subject line in imperative mood starting with a capital letter. Also see https://commit.style/ or http://chris.beams.io/posts/git-commit/ as a helpful guide how to write good commit messages.

**Comment by grisu48** [comment]
Hi Oliver, thank you for your input. I'll improve the PR accordingly and let you know, when ready :-)

**Comment by grisu48** [comment]
Thank you for the feedback @okurz. I like the idea of including it in the specfile to run the test at build time and look into that.

**Comment by dzedro** [comment]
> The whole test script looks like it is only really testing the package gd. Did you consider a test in the spec file of the package instead? If there is a good reason why you don't want that you could put the whole test into a shell script in the data folder and call that instead. Should be simpler. Also, instead of validating the string output by passing that to perl you can just rely on exit code validation

There are progress issues for tests in openQA, great when test will go to obs. Handle that in poo not in PR.

**Comment by okurz** [comment]
@dzedro not sure how I should read your comment. So should I just close this PR and reject it? I wanted to be more constructive than that :)

**Comment by dzedro** [comment]
> @dzedro not sure how I should read your comment. So should I just close this PR and reject it? I wanted to be more constructive than that :)

@grisu48 created this PR based on poo#49994. He is not familiar with obs, stop bother 'first-time contributors' with it. If you like, convert existing console tests to use script, it can become new rule and you can push it.

**Comment by grisu48** [comment]
Travis finished successfully but for some reason the status is not propagated (check [details](https://travis-ci.org/github/os-autoinst/os-autoinst-distri-opensuse/builds/666768016)). Is there anything needed for this PR?

**Comment by okurz** [comment]
@grisu48 I can see "1 successful check" so (at least now) there is a status from travis CI

**Pull request merged**